src/scalar_store.rs

```diff
@@ -216,6 +222,33 @@ pub enum ScalarExpression<F: LurkField> {
     UInt(UInt),
 }
 
+#[cfg(not(target_arch = "wasm32"))]
+impl<Fr: LurkField> Arbitrary for ScalarExpression<Fr> {
+    type Parameters = ();
+    type Strategy = BoxedStrategy<Self>;
+
+    fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+        prop_oneof!(
+            Just(Self::Nil),
+            any::<(ScalarPtr<Fr>, ScalarPtr<Fr>)>().prop_map(|(x, y)| Self::Cons(x, y)),
+            any::<Symbol>().prop_map(|x| Self::Sym(Sym::Sym(x))),
+            any::<String>().prop_map(|x| Self::Str(x)),
+            any::<FWrap<Fr>>().prop_map(|x| Self::Num(x.0)),
+            any::<(ScalarPtr<Fr>, ScalarPtr<Fr>, ScalarPtr<Fr>)>().prop_map(
+                |(arg, body, closed_env)| {
+                    Self::Fun {
+                        arg,
+                        body,
+                        closed_env,
+                    }
+                }
+            ),
+            any::<ScalarThunk<Fr>>().prop_map(|x| Self::Thunk(x)),
+        )
```

Comment:
Any reason to exclude `Char`, `UInt`, and `Comm`?

Reply:
Ah, looks like this was just moved from elsewhere…

Reply:
Unless the goal is to exclude, could we just derive `Arbitrary` for `ScalarExpression`? I do think we might want a custom implementation, so we can support well-formed commitments (i.e. those whose preimage is known). Maybe some percentage could just be random and some well-formed (more expensive, but more useful for some purposes). I'm not saying it needs to happen here, more just trying to understand the implications and thinking through how these things work.

Reply:
We do derive this now